core/templates/services/html-length.service.ts

```diff
@@ -206,16 +205,18 @@ export class HtmlLengthService {
         return 0;
       case 'oppia-noninteractive-link':
       case 'oppia-noninteractive-skillreview': {
-        const textValueAttr = domTag.getAttribute('text-with-value');
-        const textValue = textValueAttr ? textValueAttr.slice(1, -1) : '';
-        const weight = this.calculateTextWeight(textValue, calculationType);
-        return weight;
+        const textValueAttr = domTag.getAttribute('text-with-value') || '';
+        const textValue = String(
```

Comment:
The only concern I have here (and below) is the String() cast. Why is that needed? I don't think we should have that, because if the item is not a string then that should result in some kind of error, right?

Reply:
@seanlip  It is required to use string cast because the function "escapedJsonToObj()" has defined its return type as object or string; however, it can actually return values such as Object, Array, string, number, boolean or null value, as mentioned in this [Ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse). Therefore, due to strict checking, it is necessary to handle the case of return type object even though it can never actually return an object type for the given input.

Reply:
@HardikGoyal2003 Two questions:

(a) Why does the other existing usage of this approach not require a string cast?
(b) Even given the case you mention, we should probably assert that it's a string rather than cast it (and throw an error if it's not a string), right?

Reply:
@seanlip (a)The reference you provided in [this comment](https://github.com/oppia/oppia/pull/19518#issuecomment-1962794950) is being passed as an argument in the `setData()` method of ckEditor. This method accepts 2 arguments without any type safety, allowing for the passing of any type of value. However, in the current scenario, we have already defined the type of argument.

(b) Yeah we can do that, to keep check on value!

Thanks!

Reply:
I don't think you need to (or should) do a cast. Try looking into "type narrowing" and "type guards" in typescript.

Reply:
Done!